aiosmtpd/smtp.py

```diff
@@ -234,10 +238,12 @@ def _handle_client(self):
                     continue
                 yield from method(arg)
             except Exception as error:
-                yield from self.push('500 Error: ({}) {}'.format(
-                    error.__class__.__name__, str(error)))
+                status = yield from self.handle_exception(error)
+                if status is None:
+                    status = '500 Error: ({}) {}'.format(
+                        error.__class__.__name__, str(error))
+                yield from self.push(status)
```

Comment:
What do you think about changing this a little bit.

* Don't catch the exception in `handle_exception()` but instead let it percolate up to `_handle_client()`
* Wrap the `yield from self.handle_exception(error)` in a try/except
* Inside *that* `except Exception as error` log the exception and yield the 500 status

The reason being:

* When None is a valid return value, I prefer to be explicit about it, rather than letting it get returned implicitly
* Triggering the 500 inside the nested except like it is here allows `handle_exception()` to return None to trigger the 500 error.  Given that that could be a bug in the exception handler, I'd rather not allow that possibility.

Does that make sense?